Gems/LmbrCentral/Code/Source/Builders/LuaBuilder/LuaBuilderWorker.cpp

```diff
@@ -43,12 +43,29 @@ namespace LuaBuilder
                 {
                     AZStd::string watchFolder;
                     AZ::Data::AssetInfo assetInfo;
-                    AZ::IO::Path path(dependency.m_dependencyPath);
-                    bool isLuaDependency = !path.HasExtension() || path.Extension() == ".lua" || path.Extension() == ".luac";
-                    auto sourcePath = path.ReplaceExtension(".lua");
+                    // carbonated begin
+                    AZ::IO::Path path(dependency.m_dependencyPath.starts_with("/")
+                        ? dependency.m_dependencyPath.substr(1) : dependency.m_dependencyPath); // path shoul not start from a path delimiter
```

Comment:
Sometimes there might be a backslash, typically there is not, but this is a lua strings, so there can be any characters. I believe engine process a patch with back slashes OK, it changes \ to / at some moment.

Reply:
1) updated
2) updated, removed both kind of slashes in the first position. Yes, I have checked, AP works fine with both kind of slashes inside a filepath. It can't work with the slash in the first path position (o3de specific)